# Исследование установаления параметров недвижимости в г. Санкт-Петербурге.

### 0.1 Шаг 1. Обзор данных

In [1]:
import pandas as pd
data = pd.read_csv('real_estate_data.csv', sep='\t')

print('Информация о датафрейме:')
data.shape

Информация о датафрейме:


(23699, 22)

In [2]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23699 entries, 0 to 23698
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   total_images          23699 non-null  int64  
 1   last_price            23699 non-null  float64
 2   total_area            23699 non-null  float64
 3   first_day_exposition  23699 non-null  object 
 4   rooms                 23699 non-null  int64  
 5   ceiling_height        14504 non-null  float64
 6   floors_total          23613 non-null  float64
 7   living_area           21796 non-null  float64
 8   floor                 23699 non-null  int64  
 9   is_apartment          2775 non-null   object 
 10  studio                23699 non-null  bool   
 11  open_plan             23699 non-null  bool   
 12  kitchen_area          21421 non-null  float64
 13  balcony               12180 non-null  float64
 14  locality_name         23650 non-null  object 
 15  airports_nearest   

In [3]:
display(data.head(30))

total_images  last_price  total_area first_day_exposition  rooms  \
0             20  13000000.0      108.00  2019-03-07T00:00:00      3   
1              7   3350000.0       40.40  2018-12-04T00:00:00      1   
2             10   5196000.0       56.00  2015-08-20T00:00:00      2   
3              0  64900000.0      159.00  2015-07-24T00:00:00      3   
4              2  10000000.0      100.00  2018-06-19T00:00:00      2   
5             10   2890000.0       30.40  2018-09-10T00:00:00      1   
6              6   3700000.0       37.30  2017-11-02T00:00:00      1   
7              5   7915000.0       71.60  2019-04-18T00:00:00      2   
8             20   2900000.0       33.16  2018-05-23T00:00:00      1   
9             18   5400000.0       61.00  2017-02-26T00:00:00      3   
10             5   5050000.0       39.60  2017-11-16T00:00:00      1   
11             9   3300000.0       44.00  2018-08-27T00:00:00      2   
12            10   3890000.0       54.00  2016-06-30T00:00:00      2   
13            20   3550000.0       42.80  2017-07-01T00:00:00      2   
14             1   4400000.0       36.00  2016-06-23T00:00:00      1   
15            16   4650000.0       39.00  2017-11-18T00:00:00      1   
16            11   6700000.0       82.00  2017-11-23T00:00:00      3   
17             6   4180000.0       36.00  2016-09-09T00:00:00      1   
18             8   3250000.0       31.00  2017-01-27T00:00:00      1   
19            16  14200000.0      121.00  2019-01-09T00:00:00      3   
20            12   6120000.0       80.00  2017-09-28T00:00:00      3   
21            13   3200000.0       31.60  2018-03-14T00:00:00      1   
22            20   5000000.0       58.00  2017-04-24T00:00:00      2   
23            11   2950000.0       32.00  2016-10-29T00:00:00      1   
24             8   6500000.0       97.20  2015-10-31T00:00:00      2   
25             3   6800000.0       76.00  2015-10-01T00:00:00      2   
26             6   4050000.0       60.00  2017-04-28T00:00:00      4   
27            20   7100000.0       70.00  2017-05-12T00:00:00      3   
28             8   4170000.0       44.00  2017-12-13T00:00:00      1   
29             9   8600000.0      100.00  2016-04-09T00:00:00      3   

    ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0             2.70          16.0        51.00      8          NaN  ...   
1              NaN          11.0        18.60      1          NaN  ...   
2              NaN           5.0        34.30      4          NaN  ...   
3              NaN          14.0          NaN      9          NaN  ...   
4             3.03          14.0        32.00     13          NaN  ...   
5              NaN          12.0        14.40      5          NaN  ...   
6              NaN          26.0        10.60      6          NaN  ...   
7              NaN          24.0          NaN     22          NaN  ...   
8              NaN          27.0        15.43     26          NaN  ...   
9             2.50           9.0        43.60      7          NaN  ...   
10            2.67          12.0        20.30      3          NaN  ...   
11             NaN           5.0        31.00      4        False  ...   
12             NaN           5.0        30.00      5          NaN  ...   
13            2.56           5.0        27.00      5          NaN  ...   
14             NaN           6.0        17.00      1          NaN  ...   
15             NaN          14.0        20.50      5          NaN  ...   
16            3.05           5.0        55.60      1          NaN  ...   
17             NaN          17.0        16.50      7          NaN  ...   
18            2.50           5.0        19.40      2          NaN  ...   
19            2.75          16.0        76.00      8          NaN  ...   
20            2.70          27.0        48.00     11          NaN  ...   
21             NaN           5.0        16.90      2          NaN  ...   
22            2.75          25.0        30.00     15          NaN  ..

 ### 0.2 Шаг 2.1 Заполнение пропусков.

In [4]:
data.isna().sum()

total_images                0
last_price                  0
total_area                  0
first_day_exposition        0
rooms                       0
ceiling_height           9195
floors_total               86
living_area              1903
floor                       0
is_apartment            20924
studio                      0
open_plan                   0
kitchen_area             2278
balcony                 11519
locality_name              49
airports_nearest         5542
cityCenters_nearest      5519
parks_around3000         5518
parks_nearest           15620
ponds_around3000         5518
ponds_nearest           14589
days_exposition          3181
dtype: int64

<div class="alert alert-info">
<b>Комментарий студента:</b>  <i>(Ответ на 1-ый вопрос пункта №1, шага №2)</i>
<br>
Пропуски обнаружены в столбцах: days_employed, total_income (причём в равных количествах). Также приведены сведения о количестве пропущенных значений в обоих столбцах (ещё и попарно, во всех строках) и таблицы со всеми строками, где есть пропущенные значения.<br>
<br>
Можно сделать вывод, что единственный элемент в датафрейме, который обозначает пропущенные значения - это элемент NaN.     
<br>
</div>

In [5]:
data.columns

Index(['total_images', 'last_price', 'total_area', 'first_day_exposition',
       'rooms', 'ceiling_height', 'floors_total', 'living_area', 'floor',
       'is_apartment', 'studio', 'open_plan', 'kitchen_area', 'balcony',
       'locality_name', 'airports_nearest', 'cityCenters_nearest',
       'parks_around3000', 'parks_nearest', 'ponds_around3000',
       'ponds_nearest', 'days_exposition'],
      dtype='object')

In [6]:
data.count()

total_images            23699
last_price              23699
total_area              23699
first_day_exposition    23699
rooms                   23699
ceiling_height          14504
floors_total            23613
living_area             21796
floor                   23699
is_apartment             2775
studio                  23699
open_plan               23699
kitchen_area            21421
balcony                 12180
locality_name           23650
airports_nearest        18157
cityCenters_nearest     18180
parks_around3000        18181
parks_nearest            8079
ponds_around3000        18181
ponds_nearest            9110
days_exposition         20518
dtype: int64

In [7]:
null_columns = []

#for value in data.columns: #data.columns: #data.count(): 
#    if data[value].count() != data.shape[0]:
#        null_columns.append(value)
        
#null_columns     '', 

    #display(first_part_data[data[item].isna()])
    #display(second_part_data[data[item].isna()])

for value in data.columns: #data.columns: #data.count(): 
    if data[value].count() != data.shape[0]:
        null_columns.append(value)
        
null_columns

['ceiling_height',
 'floors_total',
 'living_area',
 'is_apartment',
 'kitchen_area',
 'balcony',
 'locality_name',
 'airports_nearest',
 'cityCenters_nearest',
 'parks_around3000',
 'parks_nearest',
 'ponds_around3000',
 'ponds_nearest',
 'days_exposition']

In [8]:
data[data['ceiling_height'].isna()].head()

total_images  last_price  total_area first_day_exposition  rooms  \
1             7   3350000.0        40.4  2018-12-04T00:00:00      1   
2            10   5196000.0        56.0  2015-08-20T00:00:00      2   
3             0  64900000.0       159.0  2015-07-24T00:00:00      3   
5            10   2890000.0        30.4  2018-09-10T00:00:00      1   
6             6   3700000.0        37.3  2017-11-02T00:00:00      1   

   ceiling_height  floors_total  living_area  floor is_apartment  ...  \
1             NaN          11.0         18.6      1          NaN  ...   
2             NaN           5.0         34.3      4          NaN  ...   
3             NaN          14.0          NaN      9          NaN  ...   
5             NaN          12.0         14.4      5          NaN  ...   
6             NaN          26.0         10.6      6          NaN  ...   

   kitchen_area  balcony              locality_name  airports_nearest  \
1          11.0      2.0             посёлок Шушары           12817.0   
2           8.3      0.0            Санкт-Петербург           21741.0   
3           NaN      0.0            Санкт-Петербург           28098.0   
5           9.1      NaN  городской посёлок Янино-1               NaN   
6          14.4      1.0          посёлок Парголово           52996.0   

  cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
1             18603.0               0.0            NaN               0.0   
2             13933.0               1.0           90.0               2.0   
3              6800.0               2.0           84.0               3.0   
5                 NaN               NaN            NaN               NaN   
6             19143.0               0.0            NaN               0.0   

   ponds_nearest  days_exposition  
1            NaN             81.0  
2          574.0            558.0  
3          234.0            424.0  
5            NaN             55.0  
6            NaN            155.0  

[5 rows x 22 columns]

In [9]:
data[data['floors_total'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
186              12  11640000.0        65.2  2018-10-02T00:00:00      2   
237               4   2438033.0        28.1  2016-11-23T00:00:00      1   
457               4   9788348.0        70.8  2015-08-01T00:00:00      2   
671               4   6051191.0        93.6  2017-04-06T00:00:00      3   
1757              5   3600000.0        39.0  2017-04-22T00:00:00      1   
...             ...         ...         ...                  ...    ...   
22542             5   8500000.0        63.5  2017-05-24T00:00:00      2   
22656             4   4574160.0        64.5  2017-04-02T00:00:00      2   
22808             0  14569263.0       110.4  2016-11-20T00:00:00      3   
23590             0  21187872.0       123.3  2017-04-25T00:00:00      3   
23658             6   3063600.0        43.8  2016-11-28T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
186               NaN           NaN        30.80      4          NaN  ...   
237               NaN           NaN        20.75      1          NaN  ...   
457               NaN           NaN        38.40     12          NaN  ...   
671               NaN           NaN        47.10      8          NaN  ...   
1757              NaN           NaN          NaN      9          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
22542             2.8           NaN          NaN      3          NaN  ...   
22656             NaN           NaN        31.70     20          NaN  ...   
22808             NaN           NaN        45.38      6          NaN  ...   
23590             NaN           NaN        50.40     18          NaN  ...   
23658             2.7           NaN        14.00      8          NaN  ...   

       kitchen_area  balcony    locality_name  airports_nearest  \
186           12.00      NaN  Санкт-Петербург           39197.0   
237             NaN      0.0  Санкт-Петербург           22041.0   
457           10.63      0.0  Санкт-Петербург           37364.0   
671           16.80      0.0  Санкт-Петербург           22041.0   
1757            NaN      0.0  Санкт-Петербург           22735.0   
...             ...      ...              ...               ...   
22542           NaN      NaN  Санкт-Петербург           51340.0   
22656         14.40      0.0  Санкт-Петербург           22041.0   
22808         23.42      0.0  Санкт-Петербург           19095.0   
23590         23.60      0.0  Санкт-Петербург           19095.0   
23658         15.50      2.0  Санкт-Петербург            8426.0   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
186               12373.0               1.0          123.0               0.0   
237               17369.0               0.0            NaN               1.0   
457                8322.0               2.0          309.0               2.0   
671               17369.0               0.0            NaN               1.0   
1757              11618.0               1.0          835.0               1.0   
...                   ...               ...            ...               ...   
22542             15363.0               0.0            NaN               1.0   
22656             17369.0               0.0            NaN               1.0   
22808              4529.0               0.0            NaN               0.0   
23590              4529.0               0.0            NaN               0.0   
23658             12082.0               2.0           24.0               1.0   

       ponds_nearest  days_exposition  
186              NaN             49.0  
237            374.0            251.0  
457            706.0            727.0  
671            374.0            123.0  
1757           652.0             77.0  
...              ...              ...  
22542          853.0            512.0  
22656          374.0            127.0  
22808            NaN            260.0  
23590          

In [10]:
data[data['living_area'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
3                 0  64900000.0      159.00  2015-07-24T00:00:00      3   
7                 5   7915000.0       71.60  2019-04-18T00:00:00      2   
30               12   2200000.0       32.80  2018-02-19T00:00:00      1   
37               10   1990000.0       45.80  2017-10-28T00:00:00      2   
44               13   5350000.0       40.00  2018-11-18T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23642            14   5950000.0       80.30  2018-07-03T00:00:00      3   
23663            12   6400000.0       88.00  2018-01-12T00:00:00      4   
23670             9   2450000.0       45.60  2018-02-08T00:00:00      1   
23679             0   2500000.0       35.00  2017-12-08T00:00:00      1   
23697            13  11475000.0       76.75  2017-03-28T00:00:00      2   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
3                 NaN          14.0          NaN      9          NaN  ...   
7                 NaN          24.0          NaN     22          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
37               2.50           5.0          NaN      1          NaN  ...   
44                NaN          22.0          NaN      3          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23642            2.70          12.0          NaN      5          NaN  ...   
23663            2.55          10.0          NaN      1          NaN  ...   
23670            2.80           9.0          NaN      2          NaN  ...   
23679             NaN           9.0          NaN      4          NaN  ...   
23697            3.00          17.0          NaN     12          NaN  ...   

       kitchen_area  balcony                        locality_name  \
3               NaN      0.0                      Санкт-Петербург   
7              18.9      2.0                      Санкт-Петербург   
30              NaN      NaN                             Коммунар   
37              NaN      NaN  поселок городского типа Красный Бор   
44              NaN      1.0                      Санкт-Петербург   
...             ...      ...                                  ...   
23642           NaN      NaN                              Колпино   
23663           NaN      NaN                      Санкт-Петербург   
23670           NaN      NaN     поселок городского типа Синявино   
23679           NaN      NaN                         Сосновый Бор   
23697          23.3      2.0                      Санкт-Петербург   

       airports_nearest cityCenters_nearest  parks_around3000  parks_nearest  \
3               28098.0              6800.0               2.0           84.0   
7               23982.0             11634.0               0.0            NaN   
30                  NaN                 NaN               NaN            NaN   
37                  NaN                 NaN               NaN            NaN   
44              30471.0             11603.0               1.0          620.0   
...                 ...                 ...               ...            ...   
23642           27933.0             33719.0               0.0            NaN   
23663           33642.0             13943.0               0.0            NaN   
23670               NaN                 NaN               NaN            NaN   
23679               NaN                 NaN               NaN            NaN   
23697           39140.0             10364.0               2.0          173.0   

       ponds_around3000  ponds_nearest  days_exposition  
3                   3.0          234.0            424.0  
7                   0.0            NaN              NaN  
30                  NaN            NaN             63.0  
37                  NaN            NaN            196.0  
44                  1.0         1152.0              NaN  
...                 ...            ...     

In [11]:
data[data['is_apartment'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
0                20  13000000.0      108.00  2019-03-07T00:00:00      3   
1                 7   3350000.0       40.40  2018-12-04T00:00:00      1   
2                10   5196000.0       56.00  2015-08-20T00:00:00      2   
3                 0  64900000.0      159.00  2015-07-24T00:00:00      3   
4                 2  10000000.0      100.00  2018-06-19T00:00:00      2   
...             ...         ...         ...                  ...    ...   
23694             9   9700000.0      133.81  2017-03-21T00:00:00      3   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23697            13  11475000.0       76.75  2017-03-28T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0                2.70          16.0         51.0      8          NaN  ...   
1                 NaN          11.0         18.6      1          NaN  ...   
2                 NaN           5.0         34.3      4          NaN  ...   
3                 NaN          14.0          NaN      9          NaN  ...   
4                3.03          14.0         32.0     13          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23694            3.70           5.0         73.3      3          NaN  ...   
23695             NaN           5.0         38.0      4          NaN  ...   
23696             NaN           3.0         29.7      1          NaN  ...   
23697            3.00          17.0          NaN     12          NaN  ...   
23698            2.50           5.0         12.3      1          NaN  ...   

       kitchen_area  balcony        locality_name  airports_nearest  \
0             25.00      NaN      Санкт-Петербург           18863.0   
1             11.00      2.0       посёлок Шушары           12817.0   
2              8.30      0.0      Санкт-Петербург           21741.0   
3               NaN      0.0      Санкт-Петербург           28098.0   
4             41.00      NaN      Санкт-Петербург           31856.0   
...             ...      ...                  ...               ...   
23694         13.83      NaN      Санкт-Петербург           24665.0   
23695          8.50      NaN                Тосно               NaN   
23696           NaN      NaN     село Рождествено               NaN   
23697         23.30      2.0      Санкт-Петербург           39140.0   
23698          9.00      NaN  поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0                 16028.0               1.0          482.0               2.0   
1                 18603.0               0.0            NaN               0.0   
2                 13933.0               1.0           90.0               2.0   
3                  6800.0               2.0           84.0               3.0   
4                  8098.0               2.0          112.0               1.0   
...                   ...               ...            ...               ...   
23694              4232.0               1.0          796.0               3.0   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23697             10364.0               2.0          173.0               3.0   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
0              755.0              NaN  
1                NaN             81.0  
2              574.0            558.0  
3              234.0            424.0  
4               48.0            121.0  
...              ...              ...  
23694          381.0              NaN  
23695            NaN             45.0  
23696  

In [12]:
data[data['kitchen_area'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
3                 0  64900000.0       159.0  2015-07-24T00:00:00      3   
30               12   2200000.0        32.8  2018-02-19T00:00:00      1   
37               10   1990000.0        45.8  2017-10-28T00:00:00      2   
44               13   5350000.0        40.0  2018-11-18T00:00:00      1   
59               15   6300000.0        46.5  2017-03-06T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23655             1   2414306.0        40.5  2016-09-08T00:00:00      1   
23663            12   6400000.0        88.0  2018-01-12T00:00:00      4   
23670             9   2450000.0        45.6  2018-02-08T00:00:00      1   
23679             0   2500000.0        35.0  2017-12-08T00:00:00      1   
23696            18   2500000.0        56.7  2018-02-11T00:00:00      2   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
3                 NaN          14.0          NaN      9          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
37               2.50           5.0          NaN      1          NaN  ...   
44                NaN          22.0          NaN      3          NaN  ...   
59                NaN          13.0          NaN     13          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23655            2.56          18.0         19.6     10        False  ...   
23663            2.55          10.0          NaN      1          NaN  ...   
23670            2.80           9.0          NaN      2          NaN  ...   
23679             NaN           9.0          NaN      4          NaN  ...   
23696             NaN           3.0         29.7      1          NaN  ...   

       kitchen_area  balcony                        locality_name  \
3               NaN      0.0                      Санкт-Петербург   
30              NaN      NaN                             Коммунар   
37              NaN      NaN  поселок городского типа Красный Бор   
44              NaN      1.0                      Санкт-Петербург   
59              NaN      NaN                      Санкт-Петербург   
...             ...      ...                                  ...   
23655           NaN      NaN                           Никольское   
23663           NaN      NaN                      Санкт-Петербург   
23670           NaN      NaN     поселок городского типа Синявино   
23679           NaN      NaN                         Сосновый Бор   
23696           NaN      NaN                     село Рождествено   

       airports_nearest cityCenters_nearest  parks_around3000  parks_nearest  \
3               28098.0              6800.0               2.0           84.0   
30                  NaN                 NaN               NaN            NaN   
37                  NaN                 NaN               NaN            NaN   
44              30471.0             11603.0               1.0          620.0   
59              11125.0             14501.0               0.0            NaN   
...                 ...                 ...               ...            ...   
23655               NaN                 NaN               NaN            NaN   
23663           33642.0             13943.0               0.0            NaN   
23670               NaN                 NaN               NaN            NaN   
23679               NaN                 NaN               NaN            NaN   
23696               NaN                 NaN               NaN            NaN   

       ponds_around3000  ponds_nearest  days_exposition  
3                   3.0          234.0            424.0  
30                  NaN            NaN             63.0  
37                  NaN            NaN            196.0  
44                  1.0         1152.0              NaN  
59                  0.0            NaN            336.0  
...                 ...            ...     

In [13]:
data[data['balcony'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
0                20  13000000.0      108.00  2019-03-07T00:00:00      3   
4                 2  10000000.0      100.00  2018-06-19T00:00:00      2   
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
10                5   5050000.0       39.60  2017-11-16T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23694             9   9700000.0      133.81  2017-03-21T00:00:00      3   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0                2.70          16.0        51.00      8          NaN  ...   
4                3.03          14.0        32.00     13          NaN  ...   
5                 NaN          12.0        14.40      5          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
10               2.67          12.0        20.30      3          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23694            3.70           5.0        73.30      3          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
0             25.00      NaN            Санкт-Петербург           18863.0   
4             41.00      NaN            Санкт-Петербург           31856.0   
5              9.10      NaN  городской посёлок Янино-1               NaN   
8              8.81      NaN             посёлок Мурино               NaN   
10             8.50      NaN            Санкт-Петербург           38357.0   
...             ...      ...                        ...               ...   
23692          6.00      NaN                     Тихвин               NaN   
23694         13.83      NaN            Санкт-Петербург           24665.0   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0                 16028.0               1.0          482.0               2.0   
4                  8098.0               2.0          112.0               1.0   
5                     NaN               NaN            NaN               NaN   
8                     NaN               NaN            NaN               NaN   
10                13878.0               1.0          310.0               2.0   
...                   ...               ...            ...               ...   
23692                 NaN               NaN            NaN               NaN   
23694              4232.0               1.0          796.0               3.0   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
0              755.0              NaN  
4               48.0            121.0  
5                NaN             55.0  
8                NaN            189.0  
10             553.0            137.0  
...              ...              ...  
23692          

In [14]:
data[data['locality_name'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
1097              3   8600000.0       81.70  2016-04-15T00:00:00      3   
2033              6   5398000.0       80.00  2017-05-30T00:00:00      3   
2603             20   3351765.0       42.70  2015-09-20T00:00:00      1   
2632              2   5130593.0       62.40  2015-10-11T00:00:00      2   
3574             10   4200000.0       46.50  2016-05-28T00:00:00      2   
4151             17  17600000.0       89.50  2014-12-09T00:00:00      2   
4189              7   9200000.0       80.00  2015-12-10T00:00:00      3   
4670              1   5500000.0       83.00  2015-08-14T00:00:00      3   
5343             19  13540000.0       85.50  2016-01-20T00:00:00      3   
5707              7   3700000.0       30.00  2016-04-29T00:00:00      1   
6765             20   4895892.0       60.70  2015-03-12T00:00:00      2   
7114              5   4250000.0       56.00  2016-03-16T00:00:00      3   
7330              8   5100000.0       63.00  2015-01-27T00:00:00      3   
7600              8   6800000.0       70.00  2016-01-31T00:00:00      3   
8568             10  16000000.0      155.00  2016-05-09T00:00:00      3   
8986             10   4850000.0      103.10  2018-07-10T00:00:00      3   
9821             13   8000000.0       94.50  2015-01-21T00:00:00      4   
10122             5   8200000.0       83.00  2015-06-24T00:00:00      4   
11248            12   6300000.0       63.10  2015-01-16T00:00:00      4   
12879            12   4400000.0       39.20  2016-04-26T00:00:00      1   
12936             6   6800000.0       73.00  2015-11-01T00:00:00      3   
13223             1   2919911.0       29.40  2015-03-12T00:00:00      1   
13690             7   3500000.0       71.00  2016-06-23T00:00:00      3   
14273             2   4422000.0       60.00  2016-03-23T00:00:00      2   
14342             3   3611000.0       53.50  2017-04-27T00:00:00      1   
15686            13   4700000.0       44.00  2015-12-01T00:00:00      2   
15866            10   3950000.0       44.00  2016-04-16T00:00:00      2   
16499             2   4995573.0       56.90  2016-06-17T00:00:00      2   
16561             3   2450000.0       30.00  2016-06-02T00:00:00      1   
16610            11  11940000.0      112.00  2015-11-19T00:00:00      3   
17535             2   5985000.0       79.80  2018-07-30T00:00:00      3   
17764             9   8400000.0       94.00  2016-01-24T00:00:00      3   
18526             3  10800000.0       86.00  2016-06-24T00:00:00      4   
18917             3   2660000.0       37.99  2017-08-17T00:00:00      1   
19045             6   4650000.0       48.00  2016-01-25T00:00:00      2   
19972            20   4361004.0       62.40  2015-09-20T00:00:00      2   
20057            13  11500000.0      102.00  2015-10-14T00:00:00      2   
20382             8   1750000.0       72.90  2018-10-27T00:00:00      3   
20590             7   3380000.0       56.00  2017-11-06T00:00:00      2   
20654             7   6100000.0       43.00  2016-01-13T00:00:00      1   
21119             8   3500000.0       43.20  2018-11-11T00:00:00      2   
21276             0  17122148.0      178.30  2017-02-10T00:00:00      1   
21333            10   5900000.0       58.00  2015-03-12T00:00:00      3   
21715             2   6047550.0       80.10  2018-07-30T00:00:00      2   
21898             2   5886750.0       83.50  2018-07-30T00:00:00      2   
22474             7  24000000.0      128.00  2015-07-24T00:00:00      4   
22717             9   3000000.0       35.00  2018-01-02T00:00:00      1   
22933            20   3176015.0       33.30  2015-04-22T00:00:00      1   
23214             3   7990000.0       56.00  2016-05-31T00:00:00      2   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
1097             3.55           5.0        50.80      2          NaN  ...   
2033              NaN           4.0        42.60      2          NaN  ...   
2603              NaN    

In [15]:
data[data['airports_nearest'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
12               10   3890000.0       54.00  2016-06-30T00:00:00      2   
22               20   5000000.0       58.00  2017-04-24T00:00:00      2   
30               12   2200000.0       32.80  2018-02-19T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23683            16   2100000.0       62.80  2018-09-18T00:00:00      4   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
5                 NaN          12.0        14.40      5          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
12                NaN           5.0        30.00      5          NaN  ...   
22               2.75          25.0        30.00     15          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23683            2.50           5.0        45.50      3          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
5              9.10      NaN  городской посёлок Янино-1               NaN   
8              8.81      NaN             посёлок Мурино               NaN   
12             9.00      0.0                  Сертолово               NaN   
22            11.00      2.0            деревня Кудрово               NaN   
30              NaN      NaN                   Коммунар               NaN   
...             ...      ...                        ...               ...   
23683          5.50      NaN       посёлок Дзержинского               NaN   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
5                     NaN               NaN            NaN               NaN   
8                     NaN               NaN            NaN               NaN   
12                    NaN               NaN            NaN               NaN   
22                    NaN               NaN            NaN               NaN   
30                    NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23683                 NaN               NaN            NaN               NaN   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
5                NaN             55.0  
8                NaN            189.0  
12               NaN             90.0  
22               NaN             60.0  
30               NaN             63.0  
...              ...              ...  
23683          

In [16]:
data[data['cityCenters_nearest'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
12               10   3890000.0       54.00  2016-06-30T00:00:00      2   
22               20   5000000.0       58.00  2017-04-24T00:00:00      2   
30               12   2200000.0       32.80  2018-02-19T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23683            16   2100000.0       62.80  2018-09-18T00:00:00      4   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
5                 NaN          12.0        14.40      5          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
12                NaN           5.0        30.00      5          NaN  ...   
22               2.75          25.0        30.00     15          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23683            2.50           5.0        45.50      3          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
5              9.10      NaN  городской посёлок Янино-1               NaN   
8              8.81      NaN             посёлок Мурино               NaN   
12             9.00      0.0                  Сертолово               NaN   
22            11.00      2.0            деревня Кудрово               NaN   
30              NaN      NaN                   Коммунар               NaN   
...             ...      ...                        ...               ...   
23683          5.50      NaN       посёлок Дзержинского               NaN   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
5                     NaN               NaN            NaN               NaN   
8                     NaN               NaN            NaN               NaN   
12                    NaN               NaN            NaN               NaN   
22                    NaN               NaN            NaN               NaN   
30                    NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23683                 NaN               NaN            NaN               NaN   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
5                NaN             55.0  
8                NaN            189.0  
12               NaN             90.0  
22               NaN             60.0  
30               NaN             63.0  
...              ...              ...  
23683          

In [17]:
data[data['parks_around3000'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
12               10   3890000.0       54.00  2016-06-30T00:00:00      2   
22               20   5000000.0       58.00  2017-04-24T00:00:00      2   
30               12   2200000.0       32.80  2018-02-19T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23683            16   2100000.0       62.80  2018-09-18T00:00:00      4   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
5                 NaN          12.0        14.40      5          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
12                NaN           5.0        30.00      5          NaN  ...   
22               2.75          25.0        30.00     15          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23683            2.50           5.0        45.50      3          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
5              9.10      NaN  городской посёлок Янино-1               NaN   
8              8.81      NaN             посёлок Мурино               NaN   
12             9.00      0.0                  Сертолово               NaN   
22            11.00      2.0            деревня Кудрово               NaN   
30              NaN      NaN                   Коммунар               NaN   
...             ...      ...                        ...               ...   
23683          5.50      NaN       посёлок Дзержинского               NaN   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
5                     NaN               NaN            NaN               NaN   
8                     NaN               NaN            NaN               NaN   
12                    NaN               NaN            NaN               NaN   
22                    NaN               NaN            NaN               NaN   
30                    NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23683                 NaN               NaN            NaN               NaN   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
5                NaN             55.0  
8                NaN            189.0  
12               NaN             90.0  
22               NaN             60.0  
30               NaN             63.0  
...              ...              ...  
23683          

In [18]:
data[data['parks_nearest'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
1                 7   3350000.0       40.40  2018-12-04T00:00:00      1   
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
6                 6   3700000.0       37.30  2017-11-02T00:00:00      1   
7                 5   7915000.0       71.60  2019-04-18T00:00:00      2   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23691            11   9470000.0       72.90  2016-10-13T00:00:00      2   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
1                 NaN          11.0        18.60      1          NaN  ...   
5                 NaN          12.0        14.40      5          NaN  ...   
6                 NaN          26.0        10.60      6          NaN  ...   
7                 NaN          24.0          NaN     22          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23691            2.75          25.0        40.30      7          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
1             11.00      2.0             посёлок Шушары           12817.0   
5              9.10      NaN  городской посёлок Янино-1               NaN   
6             14.40      1.0          посёлок Парголово           52996.0   
7             18.90      2.0            Санкт-Петербург           23982.0   
8              8.81      NaN             посёлок Мурино               NaN   
...             ...      ...                        ...               ...   
23691         10.60      1.0            Санкт-Петербург           19424.0   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
1                 18603.0               0.0            NaN               0.0   
5                     NaN               NaN            NaN               NaN   
6                 19143.0               0.0            NaN               0.0   
7                 11634.0               0.0            NaN               0.0   
8                     NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23691              4489.0               0.0            NaN               1.0   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
1                NaN             81.0  
5                NaN             55.0  
6                NaN            155.0  
7                NaN              NaN  
8                NaN            189.0  
...              ...              ...  
23691          

In [19]:
data[data['ponds_around3000'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
12               10   3890000.0       54.00  2016-06-30T00:00:00      2   
22               20   5000000.0       58.00  2017-04-24T00:00:00      2   
30               12   2200000.0       32.80  2018-02-19T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23683            16   2100000.0       62.80  2018-09-18T00:00:00      4   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
5                 NaN          12.0        14.40      5          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
12                NaN           5.0        30.00      5          NaN  ...   
22               2.75          25.0        30.00     15          NaN  ...   
30                NaN           9.0          NaN      2          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23683            2.50           5.0        45.50      3          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698            2.50           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
5              9.10      NaN  городской посёлок Янино-1               NaN   
8              8.81      NaN             посёлок Мурино               NaN   
12             9.00      0.0                  Сертолово               NaN   
22            11.00      2.0            деревня Кудрово               NaN   
30              NaN      NaN                   Коммунар               NaN   
...             ...      ...                        ...               ...   
23683          5.50      NaN       посёлок Дзержинского               NaN   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
5                     NaN               NaN            NaN               NaN   
8                     NaN               NaN            NaN               NaN   
12                    NaN               NaN            NaN               NaN   
22                    NaN               NaN            NaN               NaN   
30                    NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23683                 NaN               NaN            NaN               NaN   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
5                NaN             55.0  
8                NaN            189.0  
12               NaN             90.0  
22               NaN             60.0  
30               NaN             63.0  
...              ...              ...  
23683          

In [20]:
data[data['ponds_nearest'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
1                 7   3350000.0       40.40  2018-12-04T00:00:00      1   
5                10   2890000.0       30.40  2018-09-10T00:00:00      1   
6                 6   3700000.0       37.30  2017-11-02T00:00:00      1   
7                 5   7915000.0       71.60  2019-04-18T00:00:00      2   
8                20   2900000.0       33.16  2018-05-23T00:00:00      1   
...             ...         ...         ...                  ...    ...   
23690             3   5500000.0       52.00  2018-07-19T00:00:00      2   
23692             2   1350000.0       30.00  2017-07-07T00:00:00      1   
23695            14   3100000.0       59.00  2018-01-15T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
1                 NaN          11.0        18.60      1          NaN  ...   
5                 NaN          12.0        14.40      5          NaN  ...   
6                 NaN          26.0        10.60      6          NaN  ...   
7                 NaN          24.0          NaN     22          NaN  ...   
8                 NaN          27.0        15.43     26          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23690             NaN           5.0        31.00      2          NaN  ...   
23692             NaN           5.0        17.50      4          NaN  ...   
23695             NaN           5.0        38.00      4          NaN  ...   
23696             NaN           3.0        29.70      1          NaN  ...   
23698             2.5           5.0        12.30      1          NaN  ...   

       kitchen_area  balcony              locality_name  airports_nearest  \
1             11.00      2.0             посёлок Шушары           12817.0   
5              9.10      NaN  городской посёлок Янино-1               NaN   
6             14.40      1.0          посёлок Парголово           52996.0   
7             18.90      2.0            Санкт-Петербург           23982.0   
8              8.81      NaN             посёлок Мурино               NaN   
...             ...      ...                        ...               ...   
23690          6.00      NaN            Санкт-Петербург           20151.0   
23692          6.00      NaN                     Тихвин               NaN   
23695          8.50      NaN                      Тосно               NaN   
23696           NaN      NaN           село Рождествено               NaN   
23698          9.00      NaN        поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
1                 18603.0               0.0            NaN               0.0   
5                     NaN               NaN            NaN               NaN   
6                 19143.0               0.0            NaN               0.0   
7                 11634.0               0.0            NaN               0.0   
8                     NaN               NaN            NaN               NaN   
...                   ...               ...            ...               ...   
23690              6263.0               1.0          300.0               0.0   
23692                 NaN               NaN            NaN               NaN   
23695                 NaN               NaN            NaN               NaN   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
1                NaN             81.0  
5                NaN             55.0  
6                NaN            155.0  
7                NaN              NaN  
8                NaN            189.0  
...              ...              ...  
23690          

In [21]:
data[data['days_exposition'].isna()]

total_images  last_price  total_area first_day_exposition  rooms  \
0                20  13000000.0      108.00  2019-03-07T00:00:00      3   
7                 5   7915000.0       71.60  2019-04-18T00:00:00      2   
44               13   5350000.0       40.00  2018-11-18T00:00:00      1   
45               17   5200000.0       50.60  2018-12-02T00:00:00      2   
46               17   6600000.0       52.10  2019-01-31T00:00:00      2   
...             ...         ...         ...                  ...    ...   
23684            20  21400000.0      145.00  2018-11-02T00:00:00      4   
23685            15   2490000.0       31.00  2019-01-24T00:00:00      1   
23694             9   9700000.0      133.81  2017-03-21T00:00:00      3   
23696            18   2500000.0       56.70  2018-02-11T00:00:00      2   
23698             4   1350000.0       32.30  2017-07-21T00:00:00      1   

       ceiling_height  floors_total  living_area  floor is_apartment  ...  \
0                2.70          16.0         51.0      8          NaN  ...   
7                 NaN          24.0          NaN     22          NaN  ...   
44                NaN          22.0          NaN      3          NaN  ...   
45               2.65           9.0         30.3      7          NaN  ...   
46               2.60          24.0         29.7      9          NaN  ...   
...               ...           ...          ...    ...          ...  ...   
23684            3.00          26.0         71.4     17          NaN  ...   
23685            2.50           5.0         17.3      5          NaN  ...   
23694            3.70           5.0         73.3      3          NaN  ...   
23696             NaN           3.0         29.7      1          NaN  ...   
23698            2.50           5.0         12.3      1          NaN  ...   

       kitchen_area  balcony        locality_name  airports_nearest  \
0             25.00      NaN      Санкт-Петербург           18863.0   
7             18.90      2.0      Санкт-Петербург           23982.0   
44              NaN      1.0      Санкт-Петербург           30471.0   
45             7.00      NaN      Санкт-Петербург           30011.0   
46             8.30      2.0      Санкт-Петербург           15114.0   
...             ...      ...                  ...               ...   
23684         15.60      NaN      Санкт-Петербург           11827.0   
23685          5.60      1.0            Ломоносов           48393.0   
23694         13.83      NaN      Санкт-Петербург           24665.0   
23696           NaN      NaN     село Рождествено               NaN   
23698          9.00      NaN  поселок Новый Учхоз               NaN   

      cityCenters_nearest  parks_around3000  parks_nearest  ponds_around3000  \
0                 16028.0               1.0          482.0               2.0   
7                 11634.0               0.0            NaN               0.0   
44                11603.0               1.0          620.0               1.0   
45                12872.0               0.0            NaN               0.0   
46                12702.0               0.0            NaN               0.0   
...                   ...               ...            ...               ...   
23684             11459.0               0.0            NaN               0.0   
23685             51818.0               0.0            NaN               0.0   
23694              4232.0               1.0          796.0               3.0   
23696                 NaN               NaN            NaN               NaN   
23698                 NaN               NaN            NaN               NaN   

       ponds_nearest  days_exposition  
0              755.0              NaN  
7                NaN              NaN  
44            1152.0              NaN  
45               NaN              NaN  
46               NaN              NaN  
...              ...              ...  
23684            NaN              NaN  
23685            NaN              NaN  
23694  

### 0.2 Шаг 2.2 Приведение столбцов к нужным типам данных.

### 0.3 Шаг 3.1 Подсчёт данных и добавление в новые столбцы.

### 0.4 Шаг 4.1 Проведите исследовательский анализ данных.

## 1 Общий вывод: